src/explore-education-statistics-admin/src/pages/release/data/ReleaseApiDataSetDetailsPage.tsx

```diff
@@ -157,16 +183,96 @@ export default function ReleaseApiDataSetDetailsPage() {
     />
   ) : null;
 
-  function getDataSetStatusColour(status: DataSetStatus): TagProps['colour'] {
-    switch (status) {
-      case 'Deprecated':
-        return 'purple';
-      case 'Withdrawn':
-        return 'red';
-      default:
-        return 'blue';
+  const mappingComplete =
+    dataSet?.draftVersion?.mappingStatus &&
+    dataSet.draftVersion.mappingStatus.filtersComplete &&
+    dataSet.draftVersion.mappingStatus.locationsComplete;
+
+  const showDraftVersionTasks =
+    importingStatus !== 'importing' &&
+    (dataSet?.draftVersion?.status === 'Draft' ||
+      dataSet?.draftVersion?.status === 'Mapping');
+
+  const getImportBanner = () => {
+    if (importingStatus === 'importing') {
+      return (
+        <NotificationBanner
+          fullWidthContent
+          heading="Import in progress..."
+          title="Importing"
+        >
+          <p>This process can take a few minutes</p>
+        </NotificationBanner>
+      );
+    }
+
+    if (dataSet?.draftVersion?.status === 'Mapping') {
+      return (
+        <NotificationBanner
+          fullWidthContent
+          heading="This new API data set version has been mapped and is ready to be imported"
+          title="Action required"
+        >
+          <p>
+            If you have completed and checked the new mappings, the next step is
+            to import the changes to enable this version to be published. After
+            importing you will also be able to view the changelog, add guidance
+            notes and also preview this API data set version.
+          </p>
+          <p className="govuk-!-margin-bottom-6">
+            <strong>Please note</strong> you will not be able to make further
+            changes after importing. If you need to make any changes to the
+            mappings you will have to remove this API data set version and
+            create a new version.
+          </p>
+          <Button onClick={handleImport}>Import this data set version</Button>
+        </NotificationBanner>
+      );
     }
-  }
+
+    if (dataSet?.draftVersion?.status === 'Draft') {
+      return (
+        <NotificationBanner
+          fullWidthContent
+          heading="This new API data set version has now been mapped and imported ready for final publication"
+          role={importingStatus === 'imported' ? 'alert' : undefined}
```

Comment:
Not entirely sure, but changing the `alert` role like this might not trigger the banner to be read out. Think the alert only works if the browser is aware of the element (with `role="alert"`) before its text is updated. From the [Mozilla docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/alert_role): 

> As with all other [live regions](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions), alerts will only be announced when the content of the element with role="alert" is updated. Make sure that the element with the role is present in the page's markup first - this will "prime" the browser and screen reader to keep watching the element for changes. After this, any changes to the content will be announced. Do not try to dynamically add/generate an element with role="alert" that is already populated with the alert message you want announced - this generally does not lead to an announcement, as it is not a content change.

I'm not sure if the notification banner element ends up being unmounted from the DOM at any point, so the browser may not detect the element's contents change changing. 

Probably worth testing with a screen reader.

Reply:
It does work here because the role isn't changed dynamically - it only has the role `alert` when the import process has just finished (`finalisingStatus === 'finalised'`), when you visit the page again there isn't a `finalisingStatus` so the role isn't applied.

Reply:
Okay, will trust that you're right about this!